### PR TITLE
Add PEP8 to CI and format (python) code base

### LINKED ISSUE #177: Use PEP8 formatting and implement CI
See https://github.com/precice/tutorials/pull/153#discussion_r607138785.

# Todo

- [x] implement CI similar to https://github.com/precice/python-bindings/blob/develop/.github/workflows/check-pep8.yml
- [x] format all python codes according to PEP8

### PR DESCRIPTION
**Pull request by BenjaminRodenberg**
Closes #177 

@Reviewers: How do you like the formatting? The PEP8 restriction to 80 columns is quite restrictive, but this is just what we have to live with, I assume.

### CODE REVIEW

**Comment by MakisH** [comment]
Please wait until #184 is merged and rebase. It contains several Python files and would be nice to format them already.

**Comment by BenjaminRodenberg** [comment]
> Please wait until #184 is merged and rebase. It contains several Python files and would be nice to format them already.

I don't think that we need this additional barrier. Whether we merge this PR before #184 or after does not make a difference and the sooner we get this PR merged the better.

**Comment by uekerman** [comment]
Honestly, I think the 80 col restriction is horrible, but if you say that we cannot change this let's do it ... :see_no_evil:

File: partitioned-elastic-beam/visualize.py [hunk 1]
```diff
@@ -4,119 +4,146 @@
...
-
-			line_f1 = f1.readline()
-			for line_f2 in iter( f2.readline, " -3\n" ):# -3 indicates end of line block
-				# same node in both files (interface): write any (assuming their values are correct!!)
-				if line_f1[:13] == line_f2[:13]:
-					if line_f1[2] == "3":
-						continue
-					else:
-						# this is an interface node for both beams. write the mean of the values in beam1 and beam2
-						mean_vals = [ ( float(x)+float(y) ) / 2. for x,y in zip( [line_f1[13:25], line_f1[25:37], line_f1[37:49]], [line_f2[13:25], line_f2[25:37], line_f2[37:49]] ) ]
```
Comment (BenjaminRodenberg) [inline partitioned-elastic-beam/visualize.py:96]: Speaking of 80 columns: This line is (with 160 chars) clearly too long, but this is nothing a formatter can fix.

File: partitioned-heat-conduction/fenics/heat.py [hunk 2]
```diff
@@ -197,12 +198,14 @@ def determine_gradient(V_g, u, flux):
...
-        # Dirichlet problem reads temperature and writes flux on boundary to Neumann problem
+        # Dirichlet problem reads temperature and writes flux on boundary to
+        # Neumann problem
         determine_gradient(V_g, u_np1, flux)
         flux_x = interpolate(flux.sub(0), W)
         precice.write_data(flux_x)
     elif problem is ProblemType.NEUMANN:
-        # Neumann problem reads flux and writes temperature on boundary to Dirichlet problem
+        # Neumann problem reads flux and writes temperature on boundary to
+        # Dirichlet problem
```
Comment (BenjaminRodenberg) [inline partitioned-heat-conduction/fenics/heat.py:208]: This linebreak is not necessary imho.

File: perpendicular-flap/solid-fenics/perp-flap.py [hunk 3]
```diff
@@ -182,7 +185,8 @@ def avg(x_old, x_new, alpha):
 
 while precice.is_coupling_ongoing():
 
-    if precice.is_action_required(precice.action_write_iteration_checkpoint()):  # write checkpoint
+    if precice.is_action_required(
+            precice.action_write_iteration_checkpoint()):  # write checkpoint
```
Comment (BenjaminRodenberg) [inline perpendicular-flap/solid-fenics/perp-flap.py:189]: This statement should definitely fit in one line!

**Review by BenjaminRodenberg — COMMENTED** [review]
I looked up examples where the 80 columns force us to do stupid things. However, there are also some lines that are clearly too long. My suggestion: Let's go for 120 columns instead of 80 and see how it works. We should have some kind of restriction to avoid overly long lines.

**Review by uekerman — APPROVED** [review]
Formatting of Nutils cases looks good now

**Review by IshaanDesai — APPROVED** [review]
I checked all changes by opening the files in Pycharm and it looks good :+1:

**Pull request merged**